File: benchmarks/accuracy/fixtures/java/dataflow_path_traversal.java
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import javax.servlet.http.HttpServletRequest;

public class DownloadHandler {
    public byte[] download(HttpServletRequest request) throws IOException {
        String requested = request.getParameter("file");
        File target = new File("uploads", requested);
        try (FileInputStream in = new FileInputStream(target)) {
            return in.readAllBytes();
        }
    }
}

File: benchmarks/accuracy/fixtures/go/dataflow_path_traversal.go
package main

import (
	"net/http"
	"os"
	"path/filepath"
)

func download(w http.ResponseWriter, r *http.Request) {
	requested := r.URL.Query().Get("file")
	target := filepath.Join("uploads", requested)
	data, err := os.ReadFile(target)
	if err != nil {
		http.Error(w, "not found", http.StatusNotFound)
		return
	}
	w.Write(data)
}
